src/zenml/cli/model.py

```diff
@@ -55,345 +51,604 @@ def models(ctx: click.Context) -> None:
+            name: Name of the model to list versions for.
+            model_uri: Model URI of the model to list versions for.
+            metadata: Metadata or Tags to filter the model versions by.
+            count: Number of model versions to list.
+            order_by_date: Order by date.
+            created_after: List model versions created after this date.
+            created_before: List model versions created before this date.
         """
-        served_models = model_deployer.find_model_server(
-            service_uuid=uuid.UUID(served_model_uuid)
+        metadata = dict(metadata) if metadata else {}
+        model_versions = model_registry.list_model_versions(
+            name=name,
+            model_source_uri=model_uri,
+            metadata=ModelRegistryModelMetadata(**metadata),
+            count=count,
+            order_by_date=order_by_date,
+            created_after=created_after,
+            created_before=created_before,
         )
-        if not served_models:
-            warning(
-                f"No model with uuid: '{served_model_uuid}' could be found."
-            )
+        if not model_versions:
+            cli_utils.declare("No model versions found.")
             return
+        cli_utils.pretty_print_model_version_table(model_versions)
 
-        for line in model_deployer.get_model_server_logs(
-            served_models[0].uuid, follow=follow, tail=tail
-        ):
-            # don't pretty-print log lines that are already pretty-printed
-            if raw or line.startswith("\x1b["):
-                console.print(line, markup=False)
-            else:
-                try:
-                    console.print(line)
-                except MarkupError:
-                    console.print(line, markup=False)
+    @models.command(
+        "register-version",
+        help="Register a model version in the active model registry.",
+    )
+    @click.argument(
+        "name",
+        type=click.STRING,
+        required=True,
+    )
+    @click.option(
+        "--description",
+        "-d",
+        type=str,
+        default=None,
+        help="Description of the model version.",
+    )
+    @click.option(
+        "--metadata",
+        "-m",
+        type=(str, str),
+        default=None,
+        help="Metadata or Tags to add to the model version. can be used like: -m key1 value1 -m key2 value",
+        multiple=True,
+    )
+    @click.option(
+        "--version",
+        "-v",
+        type=str,
+        required=True,
+        default=None,
+        help="Version of the model to register.",
+    )
+    @click.option(
+        "--model-uri",
+        "-u",
+        type=str,
+        default=None,
+        help="Model URI of the model to register.",
+        required=True,
+    )
+    @click.option(
+        "--zenml-version",
+        type=str,
+        default=None,
+        help="ZenML version of the model to register.",
```

Comment:
Do you mean ZenML version, or zenml model version? Unclear whether 0.34.0, or something else you're talking about here...